core/src/main/scala/com/mesosphere/usi/core/matching/PortResource.scala

```diff
@@ -0,0 +1,223 @@
+package com.mesosphere.usi.core.matching
+
+import com.mesosphere.usi.core.ResourceUtil
+import com.mesosphere.usi.core.matching.PortResource._
+import com.mesosphere.usi.core.models.{ResourceMatchResult, ResourceRequirement, ResourceType}
+import com.mesosphere.usi.core.protos.ProtoBuilders
+import org.apache.mesos.v1.{Protos}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+case class PortResource(ports: Seq[Int], random: Random = Random) extends ResourceRequirement {
```

Comment:
It sounds like `*Resource` shouldn't be a subtype of a `*Requirement`. Shall we use a different naming?

Reply:
Please add docs.

What is the reason not to have this in `core-models`?

Reply:
@meln1k what would you propose? These classes were named e.g. `ScalarResourceRequirment` - to me that was too long :)